packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -251,21 +268,22 @@ export class S3Provider implements StorageProviderInterface {
    * @param conditions An array of conditions that must be met for the form upload to be accepted by S3..
    */
   async getSignedUrl(key: string, expiresAfter: number, conditions): Promise<SignedURLResponse> {
-    const result = await new Promise<PresignedPost>((resolve) => {
-      this.provider.createPresignedPost(
-        {
-          Bucket: this.bucket,
-          Fields: {
-            Key: key
-          },
-          Expires: expiresAfter,
-          Conditions: conditions
-        },
-        (err, data: PresignedPost) => {
-          resolve(data)
-        }
-      )
+    const Bucket = this.bucket
+    const Key = key
+    const Conditions = conditions
+    const client = this.provider
+    const result = await new Promise<PresignedPost>(async (resolve) => {
```

Comment:
Not sure we need a wrapper Promise since I believe all of the v3 functions support async/await. We should be able to just return the relevant data that comes from `await createPresignedPost`.

Reply:
Not sure about this too, but have updated the code.